test/popup.js

```diff
@@ -0,0 +1,40 @@
+const assert = require('assert');
+const { URL } = require('url');
+
+const popup = require('../popup').testing;
+const mobileToCanonicalHost = popup.mobileToCanonicalHost;
+const getAddToListPostBody = popup.getAddToListPostBody;
+const constructLoginUrl = popup.constructLoginUrl;
+const parseTitleFromUrl = popup.parseTitleFromUrl;
+const parseDefaultListId = popup.parseDefaultListId;
+
+describe('popup.js', () => {
+
+    it('converts mobile to canonical hostnames', () => {
+        assert.deepEqual(mobileToCanonicalHost(new URL('https://en.m.wikivoyage.org')), new URL('https://en.wikivoyage.org'));
+        assert.deepEqual(mobileToCanonicalHost(new URL('https://m.mediawiki.org')), new URL('https://mediawiki.org'));
+        assert.deepEqual(mobileToCanonicalHost(new URL('https://de.wikibooks.org')), new URL('https://de.wikibooks.org'));
+    });
+
+    it('parses title from /wiki/ path or index.php \'title\' param', () => {
+        assert.deepEqual(parseTitleFromUrl(new URL('https://en.wikipedia.org/wiki/Foobar')), 'Foobar');
+        assert.deepEqual(parseTitleFromUrl(new URL('https://fr.wikipedia.org/w/index.php?title=Paso_Fino')), 'Paso_Fino');
+    });
+
+    it('constructs login URLs with returnto and returntoquery params as needed', () => {
```

Comment:
One thing maybe worth testing is that `constructLoginUrl(new URL('https://en.wikipedia.org/wiki/Special:UserLogin?returnto=Curling'))` should be the same URL (although chances are if that gets called something is already borked).

Out of curiosity, why the mix of `''` and `` ` ``?